src/framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml

```diff
@@ -182,9 +183,15 @@ FocusScope {
                     return
                 }
 
-                if (event.key === Qt.Key_Enter || event.key === Qt.Key_Return
-                        || event.key === Qt.Key_Escape) {
+                if (event.key === Qt.Key_Enter || event.key === Qt.Key_Return || event.key === Qt.Key_Escape) {
                     event.accepted = true
+
+                    if (palettesPanelHeader && palettesPanelHeader.isSearchFieldFocused) {
```

Comment:
As I understand it, it probably isn't good software engineering practice to modify the behavior of the generic TextInputField to contain references to specific instances.  That's kind of like having a C++ base class do a switch based on something specific to its various subclasses rather than actually implement the behavior in the subclass.  Which actually does happen occasionally but definitely isn't recommended.

So I would assume you should actually be modifying the behavior of the search field itself somewhere else.  If there needs to be a more generic change to the behavior of the generic function to help make that work, that would make more sense to me.

Reply:
Yeah, I wouldn't do this normally, but *every currently implemented* function I found for the palette area in qml didn't handle a return/enter press, but I might have missed something. And I don't currently know how to set up C++/Qml connections. Could've missed something though, I'm sure. If no one else does anything about the failure to implement how MS3 worked, at least this gets it started for others 

If you have any pointers about that, or know someone who knows where [return] key is handled in palettes/search area, that would be great!